ilc/server/tailor/server-router.js

```diff
@@ -8,95 +7,130 @@ const {makeAppId} = require('../../common/utils');
+        }, '');
     }
 
-    #generatePageTpl = (route, apps) => {
+    getFragmentsContext() {
+        const route = this.getRoute();
+        const apps = this.#registryConfig.apps;
         let primarySlotDetected = false;
 
-        return _.reduce(route.slots, (res, slotData, slotName) => {
-            const appName = slotData.appName;
-            const appInfo = apps[appName];
-
-            if (appInfo === undefined) {
-                throw new this.errors.RouterError({message: 'Can\'t find info about app.', data: {appName}});
-            }
-
-            if (appInfo.ssr === undefined) {
-                return res;
-            }
-
-            const ssrOpts = deepmerge({}, appInfo.ssr);
+        return _.reduce(this.#getSsrSlotsList(route.slots, apps), (res, row) => {
+            const appId = row.appId;
+            const appInfo = row.appInfo;
 
-            if (typeof ssrOpts.src !== 'string') {
+            const ssrOpts = _.pick(row.appInfo.ssr, ['src', 'timeout', 'ignoreInvalidSsl']);
+            if (!ssrOpts.src || typeof ssrOpts.src !== 'string') {
                 throw new this.errors.RouterError({message: 'No url specified for fragment!', data: {appInfo}});
             }
 
             if (ssrOpts.ignoreInvalidSsl === true) {
                 ssrOpts['ignore-invalid-ssl'] = true;
             }
-
             delete ssrOpts.ignoreInvalidSsl;
 
-            const url = new URL(ssrOpts.src);
-            const fragmentName = makeAppId(appName, slotName);
-            const fragmentKind = slotData.kind || appInfo.kind;
-
-            const reqProps = {
-                basePath: route.basePath,
-                reqUrl: route.reqUrl,
-                fragmentName,
-            };
-
-            url.searchParams.append('routerProps', Buffer.from(JSON.stringify(reqProps)).toString('base64'));
-
-            if (slotData.props !== undefined || appInfo.props !== undefined) {
-                const appProps = _.merge({}, appInfo.props, slotData.props);
-                url.searchParams.append('appProps', Buffer.from(JSON.stringify(appProps)).toString('base64'));
-            }
-
+            const fragmentKind = row.kind || appInfo.kind;
             if (fragmentKind === 'primary' && primarySlotDetected === false) {
                 ssrOpts.primary = true;
                 primarySlotDetected = true;
             } else {
                 if (fragmentKind === 'primary') {
                     this.#logger.warn(
-                        `More then one primary slot "${slotName}" found for "${reqProps.reqUrl}".\n` +
+                        `More then one primary slot "${row.name}" found for "${this.#url}".\n` +
                         'Make it regular to avoid unexpected behaviour.'
                     );
                 }
-                delete ssrOpts.primary;
             }
 
-            ssrOpts.src = url.toString();
+            ssrOpts.appProps = _.merge({}, appInfo.props, row.props);
```

Comment:
It's not critical, but I propose to use everywhere `deepmerge` in the future, to not have inconsistency results and unpredictable behavior

for example https://github.com/namecheap/ilc/pull/247/files#diff-8d15e4857d951cee8de1012b2eab00c6d624677f8fd076209397dcf99710a7fdR56

Reply:
fixed